examples/contrib/oed/guides/amort.py

```diff
@@ -0,0 +1,152 @@
+
+
+class LinearModelGuide(nn.Module):
+
+    def __init__(self, d, w_sizes):
+        """
+        Guide for linear models. No amortisation happens over designs.
+        Amortisation over data is taken care of by analytic formulae for
+        linear models (heavy use of truth).
+
+        :param int d: the number of designs
+        :param dict w_sizes: map from variable string names to int.
+        """
+        super(LinearModelGuide, self).__init__()
+        # Represent each parameter group as independent Gaussian
+        # Making a weak mean-field assumption
+        # To avoid this- combine labels
+        self.tikhonov_diag = nn.Parameter(-2.*torch.ones(sum(w_sizes.values())))
+        self.scale_tril = {l: nn.Parameter(3.*torch.ones(d, p, p)) for l, p in w_sizes.items()}
+        # This registers the dict values in pytorch
+        # Await new version to use nn.ParamterDict
+        self._registered = nn.ParameterList(self.scale_tril.values())
+        self.w_sizes = w_sizes
+        self.softplus = nn.Softplus()
+
+    def forward(self, y_dict, design, target_labels):
+
+        y = torch.cat(list(y_dict.values()), dim=-1)
+        return self.linear_model_formula(y, design, target_labels)
+
+    def linear_model_formula(self, y, design, target_labels):
+
+        tikhonov_diag = torch.diag(self.softplus(self.tikhonov_diag))
+        xtx = torch.matmul(design.transpose(-1, -2), design) + tikhonov_diag
+        xtxi = rinverse(xtx, sym=True)
+        mu = rmv(xtxi, rmv(design.transpose(-1, -2), y))
+
+        # Extract sub-indices
+        mu = tensor_to_dict(self.w_sizes, mu, subset=target_labels)
+        scale_tril = {l: rtril(self.scale_tril[l]) for l in target_labels}
+
+        return mu, scale_tril
+
+    def guide(self, y_dict, design, observation_labels, target_labels):
+
+        pyro.module("ba_guide", self)
+
+        # Returns two dicts from labels -> tensors
+        mu, scale_tril = self.forward(y_dict, design, target_labels)
+
+        for l in target_labels:
+            w_dist = dist.MultivariateNormal(mu[l], scale_tril=scale_tril[l])
+            pyro.sample(l, w_dist)
+
+
+class SigmoidGuide(LinearModelGuide):
+
+    def __init__(self, d, n, w_sizes):
+        super(SigmoidGuide, self).__init__(d, w_sizes)
+        self.inverse_sigmoid_scale = nn.Parameter(torch.ones(n))
+        self.h1_weight = nn.Parameter(torch.ones(n))
+        self.h1_bias = nn.Parameter(torch.zeros(n))
+
+    def forward(self, y_dict, design, target_labels):
+
+        y = torch.cat(list(y_dict.values()), dim=-1)
+
+        # Approx invert transformation on y in expectation
+        y, y1m = y.clamp(1e-35, 1), (1.-y).clamp(1e-35, 1)
+        logited = y.log() - y1m.log()
+        y_trans = logited/.1
+        y_trans = y_trans * self.inverse_sigmoid_scale
+        hidden = self.softplus(y_trans)
+        y_trans = y_trans + hidden * self.h1_weight + self.h1_bias
+
+        return self.linear_model_formula(y_trans, design, target_labels)
```

Comment:
Based on the doc I wrote up, we can use the linear model formula replacing `y` with its expectation conditional on `\tilde{y}`. We ought to add a `y` dependent covariance term as well (missing for now)